src/NLog/Config/StackTraceUsage.cs

```diff
@@ -33,36 +33,56 @@
 
 namespace NLog.Config
 {
+    using System;
+
     /// <summary>
     /// Value indicating how stack trace should be captured when processing the log event.
     /// </summary>
+    [Flags]
     public enum StackTraceUsage
     {
         /// <summary>
         /// Stack trace should not be captured.
         /// </summary>
-        None = 0, 
+        None = 0,
+
+        /// <summary>
+        /// Stack trace should be captured
+        /// </summary>
+        WithStackTrace = 1,
+
+#if !SILVERLIGHT
+        /// <summary>
+        /// Source-level information should be capture (source-filename + linenumber)
+        /// </summary>
+        WithSourceCode = 2,
+#endif
+
+        /// <summary>
+        /// Stack trace should only be captured if callsite details are missing
+        /// </summary>
+        WithCallSite = 4,
```

Comment:
maybe WithOnlyCallSite?

Reply:
Please tell reason

Reply:
I've difficulties to understand the difference between all the options now.  I'm afraid it's too difficult for the users.

Reply:
Only highly skilled people who knows about the NLog Callsite-Engine and will be creating their own special LayoutRenderers/Targets will have to know about these enum-values. Pretty sure no-one knows how to use them right now.

Reply:
But maybe that should be in the docs. It isn't explaining now what it does (and so the difference between callsite and stackstrace is difficult)

Please note that in VS there is also a callstack window, which is 50% callsite  and 50% stackstrace  ;)

Reply:
Well in NLog callsite means the calling method (aka caller information). and stacktrace means the callstack. Both of them have the option to include-source-information (filename + linenumber). But I'm pretty sure you know all this already.

Reply:
When trying to keep things simple, I'm not the only audience ;)

Reply:
NLog callsite-name comes from the `${callsite}` layoutrenderer. Keeping this name because I guess people using NLog, are used to the NLog-names.

Reply:
> When trying to keep things simple, I'm not the only audience ;)

Well I guess there 4 people in total, when including me and you :)

Reply:
I find the enum-names very simple. And have just kept the old enum-names to ensure backward-compatible. But made it very explicit that they are combination of the new enum-flag-values.

Reply:
Updated docs, this is fine for me know. Thanks!